### PR TITLE
expose: bump service name length to DNS1123_LABEL

### PR DESCRIPTION
**Pull request by bacongobbler**
This bumps the maximum service name length to 63 characters, which is the maximum length of a
domain name label as per RFC 1123.

This is a continuation of #25041 

closes #3572

### CODE REVIEW

File: pkg/kubelet/envvars/envvars.go [hunk 1]
```diff
@@ -60,15 +66,19 @@ func FromServices(services *api.ServiceList) []api.EnvVar {
 }
 
 func makeEnvVariableName(str string) string {
-	// TODO: If we simplify to "all names are DNS1123Subdomains" this
-	// will need two tweaks:
-	//   1) Handle leading digits
-	//   2) Handle dots
-	return strings.ToUpper(strings.Replace(str, "-", "_", -1))
+	return strings.ToUpper(strings.Replace(strings.Replace(str, ".", "_", -1), "-", "_", -1))
```
Comment (dims) [inline pkg/kubelet/envvars/envvars.go]: @bacongobbler toleration of '.' and number as the first char...do they belong in another PR? (since it's not mentioned in the commit message?

Comment (bacongobbler) [inline pkg/kubelet/envvars/envvars.go]: see https://github.com/kubernetes/kubernetes/pull/25041#issuecomment-216373916

Comment (bacongobbler) [inline pkg/kubelet/envvars/envvars.go]: also notice the comment removed above that notes the tweaks necessary to this code.

Comment (dims) [inline pkg/kubelet/envvars/envvars.go]: @bacongobbler ah thanks for the context. That really helped me understand.

**Comment by zmerlynn** [comment]
(Overloaded, removing assignment for now.)

**Comment by bacongobbler** [comment]
rebased.

**Comment by bacongobbler** [comment]
closing this as the decision has been made to move to RFC 1035 instead.
